packages/flutter_tools/lib/src/commands/analyze.dart

```diff
@@ -16,6 +16,70 @@ import '../build_configuration.dart';
 import '../globals.dart';
 import '../runner/flutter_command.dart';
 
+bool _addPackage(String directoryPath, List<String> dartFiles, Set<String> pubSpecDirectories) {
+  bool foundOne = false;
+
+  // .../directoryPath/*/bin/*.dart
+  // .../directoryPath/*/lib/main.dart
+  // .../directoryPath/*/test/*_test.dart
+  // .../directoryPath/*/test/*/*_test.dart
+  // .../directoryPath/*/benchmark/*/*_bench.dart
+
+  Directory binDirectory = new Directory(path.join(directoryPath, 'bin'));
+  if (binDirectory.existsSync()) {
+    for (FileSystemEntity subentry in binDirectory.listSync()) {
+      if (subentry is File && subentry.path.endsWith('.dart')) {
```

Comment:
I added three, isDartFile, isDartTestFile, and isDartBenchmarkFile, which seemed clearer and reduced the number of string literals.